hathor/websocket/protocol.py

```diff
@@ -30,23 +34,135 @@ class HathorAdminWebsocketProtocol(WebSocketServerProtocol):
         can send the data update to the clients
     """
 
+    MAX_GAP_LIMIT: int = 30
+
     def __init__(self, factory: 'HathorAdminWebsocketFactory') -> None:
         self.log = logger.new()
         self.factory = factory
         self.subscribed_to: set[str] = set()
+        self._history_streamer: HistoryStreamer | None = None
+        self._manual_address_iter: ManualAddressSequencer | None = None
         super().__init__()
 
     def onConnect(self, request):
+        """Called by the websocket protocol when the connection is opened but it is still pending handshaking."""
         self.log.info('connection opened, starting handshake...', request=request)
 
     def onOpen(self) -> None:
+        """Called by the websocket protocol when the connection is established."""
         self.factory.on_client_open(self)
         self.log.info('connection established')
 
     def onClose(self, wasClean, code, reason):
+        """Called by the websocket protocol when the connection is closed."""
         self.factory.on_client_close(self)
         self.log.info('connection closed', reason=reason)
 
     def onMessage(self, payload: Union[bytes, str], isBinary: bool) -> None:
+        """Called by the websocket protocol when a new message is received."""
```

Comment:
Maybe limit the number of streamers per connection to prevent abuse.

Reply:
This could be implemented in nginx, which I think we already do on other `ws` endpoints.

Reply:
I'll leave it as it is for now since this API is not intended to be publicly available.